spec/zoneddatetime.html

```diff
@@ -590,7 +590,8 @@ <h1>Temporal.ZonedDateTime.prototype.with ( _temporalZonedDateTimeLike_ [ , _opt
         1. Else,
           1. Set _offsetString_ to ? Get(_fields_, *"offset"*).
         1. Let _dateTimeResult_ be ? InterpretTemporalDateTimeFields(_calendar_, _fields_, _options_).
-        1. Let _offsetNanoseconds_ be ? ParseTimeZoneOffsetString(? ToString(_offsetString_)).
+        1. Assert: Type( _offsetString_ ) is a String.
```

Comment:
how can this assertion be correct? it comes from a `Get` into user code.

Reply:
hum..... wait ... I was wrong

> Separately, I think this is worth a normative change to move the ToString before step 22.a. If we go through the code path in 23.a. then we do know for sure that _offsetString_ is a String, because PrepareTemporalFields would have already performed ToString on that value.

I think I got mislead by ptomato  on the above statement. This is not true
PrepareTemporalFields  only guarantee the "offset" to be either String or undefine, it does not guanratee "offset" to be always String.